### PR TITLE
Create an Advanced Lua Demo with 3+ Args passed through Lua Script

### LINKED ISSUE #1643: Create an Advanced Lua Demo with 3+ Args passed through Lua Script
As the custom-scripts team works towards its final demo for the quarter, we wanted to practice writing more advanced scripts that could interact with more game items and player inputs. Our goal is to create a demo that passes at least 3 parameters through at least 3 Lua scripts. This way, we can have our scripts interact with one another and perform more complex actions.

This issue is complete when we have a completed pull request which features a demonstration contained in a .c file and its corresponding .lua files. The demo should successfully run in Chiventure and pass 3+ variables through Lua scripts.

### PR DESCRIPTION
**Pull request by tatmaca**
Pull Request for Issue [#1643](https://github.com/uchicago-cs/chiventure/issues/1643):

"As the custom-scripts team works towards its final demo for the quarter, we wanted to practice writing more advanced scripts that could interact with more game items and player inputs. Our goal is to create a demo that passes at least 3 parameters through at least 3 Lua scripts. This way, we can have our scripts interact with one another and perform more complex actions.

This issue is complete when we have a completed pull request which features a demonstration contained in a .c file and its corresponding .lua files. The demo should successfully run in Chiventure and pass 3+ variables through Lua scripts."

### CODE REVIEW

File: include/custom-scripts/arg_custom_type.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+typedef enum data_type
+{
+    NONE_TYPE = 0,
+    BOOL_TYPE = 1,
+    CHAR_TYPE = 2,
+    INT_TYPE = 3,
+    STR_TYPE = 4,
+} data_type_t;
+
+typedef union data {
```
Comment (jackjackkimkim) [inline include/custom-scripts/arg_custom_type.h]: i like that you created a central data struct that will be able to take any type of data

File: src/custom-scripts/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 add_library(custom-scripts
-            src/custom_type.c)
+            src/arg_custom_type.c
```
Comment (jackjackkimkim) [inline src/custom-scripts/CMakeLists.txt]: good now the library has all the c files that resulted from the refactoring

**Review by jackjackkimkim — APPROVED** [review]
looks good! i'm not sure when we might use more than 3 arguments for LUA but if we do, now we're covered!

File: src/custom-scripts/examples/lost_order2.lua [hunk 3]
```diff
@@ -0,0 +1,13 @@
+function foo(name, order)
+  string_success = string.format("You check the name on the recipt and... it says %s! This order is yours!", name)
+  string_failure = string.format("You check the name on the recipt and... it says Borja? This order isn't yours!")
+  string_confused = "It has the wrong receipt..."
+
+  name2 = string.lower(name)
+
+  if order == "Wingstop" then
```
Comment (ecoble) [inline src/custom-scripts/examples/lost_order2.lua:8]: Please add parenthesis to these if statements.

File: tests/custom-scripts/arg_custom_type.h [hunk 4]
```diff
@@ -9,6 +9,7 @@
 #include "lauxlib.h"
```
Comment (ecoble) [inline tests/custom-scripts/arg_custom_type.h]: Same comment here about removing this file and using the existing ones in the include folder.

**Review by ecoble — CHANGES_REQUESTED** [review]

File: src/custom-scripts/examples/lost_order2.lua (hunk 3 above)
Comment (RamarkoB) [reply src/custom-scripts/examples/lost_order2.lua:8]: This is a lua file, so I don't know if we can add parenthesis and still have the file work

Comment (ecoble) [reply src/custom-scripts/examples/lost_order2.lua:8]: There are parenthesis in the other Lua file lost_order.lua

Comment (ecoble) [reply src/custom-scripts/examples/lost_order2.lua:8]: You can remove the other parenthesis if you would rather do that, please just be consistent

**Review by ecoble — CHANGES_REQUESTED** [review]
Small style fixes

**Review by ecoble — APPROVED** [review]

**Comment by RamarkoB** [comment]
We have a completed pull request which features a demonstration contained in a .c file and its corresponding .lua files. The demo should successfully run in Chiventure and pass 3+ variables through Lua scripts. The file is lost_order.c and the corresponding lua files are lost_order.lua, lost_order2.lua and lost_order3.lua. Marking this issue as closed

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Nice work on this demo, and great job reviewing!

**Pull request merged**